lib/fragment.js

```diff
@@ -183,9 +184,12 @@ module.exports = class Fragment extends EventEmitter {
             return;
         }
 
+        const range = [this.index, this.index + this.scriptRefs.length - 1];
+        const fragmentId = this.fragmentId;
+
         this.scriptRefs.forEach((uri)=> {
             const id = this.index;
-            const attributes = Object.assign({}, this.getPipeAttributes(), { id: this.attributes.id || id });
+            const attributes = Object.assign({}, this.getPipeAttributes(), { id: this.attributes.id || id, range, fragmentId });
```

Comment:
+ We should combine both id and fragmentId 
+ Use index only for script ordering
+ Id/fragmentId for all the other cases - context, perf hooks